subgraphs/across-v2/src/availableRoutesApi.ts

```diff
@@ -0,0 +1,929 @@
+  },
+  {
+    originChainId: 288,
+    originToken: "0xdc0486f8bf31df57a952bcd3c1d3e166e3d9ec8b",
+    destinationChainId: 137,
+    destinationToken: "0x1bfd67037b42cf73acf2047067bd4f2c47d9bfd6",
+  },
+  {
+    originChainId: 288,
+    originToken: "0x780f33ad21314d9a1ffb6867fe53d48a76ec0d16",
+    destinationChainId: 137,
+    destinationToken: "0x3066818837c5e6ed6601bd5a91b0762877a6b731",
+  },
+  {
+    originChainId: 288,
+    originToken: "0xf74195bb8a5cf652411867c5c2c5b8c2a402be35",
+    destinationChainId: 137,
+    destinationToken: "0x8f3cf7ad23cd3cadbd9735aff958023239c6a063",
+  },
+  {
+    originChainId: 288,
+    originToken: "0x96821b258955587069f680729cd77369c0892b40",
+    destinationChainId: 137,
+    destinationToken: "0xf328b73b6c685831f238c30a23fc19140cb4d8fc",
+  },
+  {
+    originChainId: 288,
+    originToken: "0xdeaddeaddeaddeaddeaddeaddeaddeaddead0000",
+    destinationChainId: 42161,
+    destinationToken: "0x82af49447d8a07e3bd95bd0d56f35241523fbab1",
+  },
+  {
+    originChainId: 288,
+    originToken: "0xdeaddeaddeaddeaddeaddeaddeaddeaddead0000",
+    destinationChainId: 42161,
+    destinationToken: "0x82af49447d8a07e3bd95bd0d56f35241523fbab1",
+  },
+  {
+    originChainId: 288,
+    originToken: "0x66a2a913e447d6b4bf33efbec43aaef87890fbbc",
+    destinationChainId: 42161,
+    destinationToken: "0xff970a61a04b1ca14834a43f5de4533ebddb5cc8",
+  },
+  {
+    originChainId: 288,
+    originToken: "0xdc0486f8bf31df57a952bcd3c1d3e166e3d9ec8b",
+    destinationChainId: 42161,
+    destinationToken: "0x2f2a2543b76a4166549f7aab2e75bef0aefc5b0f",
+  },
+  {
+    originChainId: 288,
+    originToken: "0x780f33ad21314d9a1ffb6867fe53d48a76ec0d16",
+    destinationChainId: 42161,
+    destinationToken: "0xd693ec944a85eeca4247ec1c3b130dca9b0c3b22",
+  },
+  {
+    originChainId: 288,
+    originToken: "0xf74195bb8a5cf652411867c5c2c5b8c2a402be35",
+    destinationChainId: 42161,
+    destinationToken: "0xda10009cbd5d07dd0cecc66161fc93d7c9000da1",
+  },
+  {
+    originChainId: 288,
+    originToken: "0x96821b258955587069f680729cd77369c0892b40",
+    destinationChainId: 42161,
+    destinationToken: "0x53691596d1bce8cea565b84d4915e69e03d9c99d",
+  },
+  {
+    originChainId: 42161,
+    originToken: "0x2f2a2543b76a4166549f7aab2e75bef0aefc5b0f",
+    destinationChainId: 1,
+    destinationToken: "0x2260fac5e5542a773aa44fbcfedf7c193bc2c599",
+  },
+  {
+    originChainId: 42161,
+    originToken: "0xff970a61a04b1ca14834a43f5de4533ebddb5cc8",
+    destinationChainId: 1,
+    destinationToken: "0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48",
+  },
+  {
+    originChainId: 42161,
+    originToken: "0x82af49447d8a07e3bd95bd0d56f35241523fbab1",
+    destinationChainId: 1,
+    destinationToken: "0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2",
+  },
+  {
+    originChainId: 42161,
+    originToken: "0x82af49447d8a07e3bd95bd0d56f35241523fbab1",
+    destinationChainId: 1,
+    destinationToken: "0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2",
```

Comment:
duplicates

Reply:
Is this object necessary? From what i see the piece we are missing is the destination token (on chain), but if any more tokens are added it will break our subgraph...

Reply:
Looks like the info is easily avail with a contract call on mainnet, and then the avail in the respective chains relayer contract as seen in this transaction: https://optimistic.etherscan.io/tx/0x6b6bdae2b28d9f7fb58b9ea59de11a2df79a9f4bb41d0d941da3134df1752d68#eventlog

Reply:
> Looks like the info is easily avail with a contract call on mainnet, and then the avail in the respective chains relayer contract as seen in this transaction: https://optimistic.etherscan.io/tx/0x6b6bdae2b28d9f7fb58b9ea59de11a2df79a9f4bb41d0d941da3134df1752d68#eventlog

1) Let's take an example to discuss. Assume, I'm handling a filled relay on optimism subgraph and I get access to originChainId, destinationChainId, destinationToken from the event log. How do I find the originToken in this case?

2) Which contract call are you referring to mainnet? I didn't find any and based on my discussion in across discord, it wasn't possible to get originToken. For convenience, contracts on mainnet: [hub pool contract](https://etherscan.io/address/0xc186fA914353c44b2E33eBE05f21846F1048bEda#readContract), [spoke pool](https://etherscan.io/address/0x4D9079Bb4165aeb4084c526a32695dCfd2F77381#readProxyContract), [adaptor](https://etherscan.io/address/0x527E872a5c3f0C7c24Fe33F2593cFB890a285084#code)

Reply:
>Is this object necessary? From what i see the piece we are missing is the destination token (on chain), but if any more tokens are added it will break our subgraph...

@jaimehgb and I had discussed this. we decided to go ahead with it but document it because it wasn't possible. if it is possible to fetch this on-chain, we can move away from this approach.

Reply:
> duplicates
I intentionally didn't do any transformation/modifications of the API response so we anybody could easily update this object when required. The API response does return duplicates.

Here's the command I use for creating the object value: `curl https://across.to/api/available-routes | jq '.[]' | pbcopy`

Reply:
- Here is the [contract call](https://etherscan.io/address/0xc186fA914353c44b2E33eBE05f21846F1048bEda#readContract#F12) on mainnet. You need the L1 token address and destination ID, and it returns the destinationToken.
- And unfortunately it doesn't look like you can get the originToken on L2s :/ 

The current solution is good! Thanks for entertaining my idea

Reply:
Feel free to resolve once you read this